sql/item.cc

```diff
@@ -2536,6 +2536,25 @@ void Item_field::set_field(Field *field_par)
   fixed= 1;
   if (field->table->s->tmp_table == SYSTEM_TMP_TABLE)
     any_privileges= 0;
+
+  if (field && context && context->select_lex &&
```

Comment:
Again, why you check `field` if code before asserts it is **non-null**. Remove all needless conditions!